### PR TITLE
Create benchmarks for battlepass

### PR DESCRIPTION
**Pull request by FiberMan**

### CODE REVIEW

**Comment by 2075** [comment]
is this PR ready to merge @FiberMan @vayesy ? It is still in draft mode.

**Comment by FiberMan** [comment]
> is this PR ready to merge @FiberMan @vayesy ? It is still in draft mode.

Ready to be merged.

File: battlepass/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        collection_id: 0
+    };
+
+    let _ = BPass::<T>::create_battlepass(RawOrigin::Signed(caller.clone()).into(), org_id, battlepass.cid.clone(), battlepass.name.clone(), battlepass.price.clone());
+
+    <T as frame_system::Config>::Hashing::hash_of(&battlepass)
+}
+
+fn activate_bpass<T: Config>(caller: T::AccountId, battlepass_id: T::Hash) {
+    let _ = BPass::<T>::activate_battlepass(RawOrigin::Signed(caller).into(), battlepass_id);
```
Comment (vayesy) [inline battlepass/src/benchmarking.rs:56]: I would suggest to check that extrinsic executed successfully. Here and for other helpers as well.

File: battlepass/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+
+benchmarks! {
+
+	create_battlepass {
+        let caller: T::AccountId = get_funded_caller::<T>()?;
+        let org_id = get_org::<T>(caller.clone());
+        let str = BoundedVec::truncate_from(vec![1,2]);
+    }: _(RawOrigin::Signed(caller), org_id, str.clone(), str.clone(), 10)
```
Comment (vayesy) [inline battlepass/src/benchmarking.rs:94]: Add `verify` statement to check for expected result. Here and for other benchmarks.

File: signal/src/lib.rs [hunk 3]
```diff
@@ -561,8 +567,6 @@ pub mod pallet {
 			// For Absolute majority if more then 50% of members vote for one option, the proposal period ends earlier.
 			if let Some(final_proposal_state) = Self::try_finalize_proposal(&voting) {
 				Self::finalize_proposal(&proposal_id, final_proposal_state, &voting)?;
-			} else {
-				log::error!("Proposal finalization failed for proposal_id: {:?}.", proposal_id);
```
Comment (vayesy) [inline signal/src/lib.rs:565]: Any reason to drop this log message?

File: battlepass/src/benchmarking.rs (hunk 2 above)
Comment (FiberMan) [reply battlepass/src/benchmarking.rs:94]: Will add it in a separate PR.

**Pull request merged**